common/persistence/cassandra/matching_task_store.go

```diff
@@ -630,6 +639,44 @@ func (d *MatchingTaskStore) ListTaskQueueUserDataEntries(ctx context.Context, re
 	return response, nil
 }
 
+func (d *MatchingTaskStore) GetTaskQueuesByBuildId(ctx context.Context, request *p.GetTaskQueuesByBuildIdRequest) ([]string, error) {
+	query := d.Session.Query(templateListTaskQueueNamesByBuildId, request.NamespaceID, request.BuildID).WithContext(ctx)
+	iter := query.PageSize(listTaskQueueNamesByBuildIdPageSize).Iter()
+
+	var taskQueues []string
+	row := make(map[string]interface{})
+
+OUTER:
+	for {
+		for iter.MapScan(row) {
+			taskQueueRaw, ok := row["task_queue_name"]
+			if !ok {
+				return nil, newFieldNotFoundError("task_queue_name", row)
+			}
+			taskQueue, ok := taskQueueRaw.(string)
+			if !ok {
+				var stringType string
+				return nil, newPersistedTypeMismatchError("task_queue_name", stringType, taskQueueRaw, row)
+			}
+
+			taskQueues = append(taskQueues, taskQueue)
+			if len(taskQueues) > request.Limit {
```

Comment:
We'll need to monitor when this happens and probably prevent another insert.

```suggestion
			if len(taskQueues) == request.Limit {
```

Reply:
Not relevant, validation will happen on write.